programs/candidate_staking/src/lib.rs

```diff
@@ -52,6 +52,8 @@ pub mod candidate_staking {
                 msg!("You can transfer");
                 msg!("Transfer is initiated");
 
+                ctx.accounts.base_account.staked_amount = amount;
```

Comment:
```
ctx.accounts.base_account.staked_amount += amount
```

Reply:
ok will change that

Reply:
check also for overflow

Reply:
we are checking the overflow in the condition before

Reply:
we are checking if is less than the max staked amount

Reply:
ok, well done